src/Umbraco.Core/Composing/LightInject/LightInjectContainer.cs

```diff
@@ -84,41 +95,52 @@ protected static ServiceContainer CreateServiceContainer()
         /// <summary>
         /// Gets the LightInject container.
         /// </summary>
-        protected ServiceContainer Container { get; }
+        //protected ServiceContainer Container => serviceContainer;
 
         /// <inheritdoc cref="IRegister"/>
         /// <inheritdoc cref="IFactory"/>
-        public object Concrete => Container;
+        public object Concrete => lightinjectContainer;
+
+        protected ServiceContainer LightinjectContainer => lightinjectContainer;
 
         /// <inheritdoc />
         public void Dispose()
         {
             if (Interlocked.Exchange(ref _disposed, 1) == 1)
                 return;
 
-            Container.Dispose();
+            lightinjectContainer.Dispose();
         }
 
         /// <inheritdoc />
         public IFactory CreateFactory() => this;
 
+        #region IServiceProvider
+
+        public object GetService(Type type)
+        {
+            return serviceProvider.GetService(type);
+        }
+
+        #endregion
+
         #region Factory
 
         /// <inheritdoc />
         public object GetInstance(Type type)
-            => Container.GetInstance(type);
+            => this.GetService(type);
 
         /// <inheritdoc />
         public object TryGetInstance(Type type)
-            => Container.TryGetInstance(type);
+            => this.GetRequiredService(type);
```

Comment:
Get and TryGet vs Get and GetRequired = should be the other way round

Reply:
I suspected so. :) Thanks.